feat(web): load original videos

andre-antunesdesa opened the pull request:
## Description

There is currently no way to load the original videos in the browser native video player for video assets if a transcoded version is available. To enable a good experience for users with limited bandwidth, it is possible to tune transcoding settings to the worst link expected for a deployment, but that may be too low for other users who have high bandwidth and want higher quality playback (e.g. lan vs high latency wan).

This PR introduces a global setting for users to select if they always want to load the original video in the native player.  There seems to be some precedence for this with a similar global setting "Display original photos". The PR also adds the ability to change the video source between original/transcoded (if it exists) of the native player in "real-time" while on the asset page with the native player. If the page is refreshed, the source falls back to the global setting for that user.

## How Has This Been Tested?

The checks in https://immich.app/docs/developer/pr-checklist/#web-checks were run on the /web folder, where the changes were made and it seems to return no errors related to my changes. This was the only error, which also happens on main:
```
 FAIL  src/lib/utils/timeline-util.spec.ts > formatGroupTitle > formats last week
AssertionError: expected 'Sun, Jul 21' to be 'Sunday' // Object.is equality

Expected: "Sunday"
Received: "Sun, Jul 21"

 ❯ src/lib/utils/timeline-util.spec.ts:37:36
     35|     const date = parseUtcDate('2024-07-21T00:00:00Z');
     36|     locale.set('en');
     37|     expect(formatGroupTitle(date)).toBe('Sunday');
       |                                    ^
     38|     locale.set('ar-SA');
     39|     expect(formatGroupTitle(date)).toBe('الأحد');
```

Additionally, the changes were tested on a devcontainer with a test instance where transcoding was turned on for a single video uploaded. The changes were also built locally from the `docker-compose.prod.yml` and used on my "production" immich instance successfully.

## Notes
~~I'm not sure how the translations are supposed to be done for PRs. I can certainly try to run through ChatGPT to get other translations if that helps.~~ See comment below about translations (I've removed a translation per request)

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation if applicable
- [x] I have no unrelated changes in the PR.
- [ ] I have confirmed that any new dependencies are strictly necessary.
- [ ] I have written tests for new code (if applicable)
- [x] I have followed naming conventions/patterns in the surrounding code
- [ ] All code in `src/services/` uses repositories implementations for database calls, filesystem operations, etc.
- [ ] All code in `src/repositories/` is pretty basic/simple and does not have any immich specific logic (that belongs in `src/services/`)

andre-antunesdesa committed:
added user preference for always loading original video

added ability to toggle between transcoded/original in the video viewer

add fix to static check error

mertalev commented on i18n/pt_BR.json:
```diff
@@ -1402,6 +1402,9 @@
   "play_or_pause_video": "Reproduzir ou Pausar vídeo",
+  "play_original_video": "Reproduzir vídeo original",
+  "play_original_video_setting_description": "Prefira reproduzir vídeo original em vez de versão transcodificada. Se o vídeo original não for compatível, poderá resultar em erros de reprodução.",
+  "play_transcoded_video": "Reproduzir vídeo transcodificado",
```
You need to do this through Weblate. Only add the en version in the PR

andre-antunesdesa replied:
I've removed the pt_BR changes and edited my PR description asking about translations, but I'm not sure I understand how to add the translation and not mess up my own deployment (which is currently from the fork with the translation that I need). Guidance would be appreciated.

andre-antunesdesa committed:
address PR comments

andre-antunesdesa commented:
> Can you rebase this PR?

@mertalev I've rebased the PR and addressed your comments, but I'm afraid the rebase broke the code as on my devcontainer test environment the original video seems to always play regardless. I don't have time to figure out what happened right now but I'll try to take a look at it at some point. If you can easily tell what has changed that could have caused it to break, I'd appreciate the guidance.
I also don't remember requesting the review from danieldietzler above so I'm hoping it was an automatic github thing (if it was something I did, sorry about that)

mertalev commented:
No worries, I can take another look at the PR tomorrow :)

danieldietzler reviewed (approved):
The code seems fine, thanks

mertalev committed:
Update asset-viewer-nav-bar.svelte

Co-authored-by: Daniel Dietzler <36593685+danieldietzler@users.noreply.github.com>

mertalev committed:
formatting
